include/fc/reflect/typename.hpp

```diff
@@ -69,10 +71,35 @@ namespace fc {
          return n.c_str();  
      } 
   };
+  template<typename E> struct get_typename< std::set<E> >
+  {
+     static const char* name()
+     {
+        static std::string n = std::string("std::set<") + std::string(get_typename<E>::name()) + std::string(">");
+        return n.c_str();
+     }
+  };
+ template<typename A, typename B> struct get_typename< std::pair<A,B> >
+  {
+      static const char* name()
+      {
+         static std::string n = std::string("std::pair<") + get_typename<A>::name() + "," + get_typename<B>::name() + ">";
+         return n.c_str();
+      }
+  }; 
+  template<typename T> struct get_typename< fc::smart_ref<T> >
+  {
+     static const char* name()
+     {
+        return (std::string("fc::smart_ref<") + get_typename<T>::name() + std::string(">")).c_str();
```

Comment:
Must assign to static string n first, then return that.

Reply:
My oversight. Thank you for catching that.